examples/preproc3-NVT_E2E.ipynb

```diff
@@ -0,0 +1,890 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.environ[\"CUDA_DEVICE_ORDER\"] = \"PCI_BUS_ID\"\n",
+    "os.environ[\"CUDA_VISIBLE_DEVICES\"] = \"3\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "evaluation = False\n",
+    "evaluation_verbose = False\n",
+    "OUTPUT_BUCKET_FOLDER = \"./preprocessed/\"\n",
+    "DATA_BUCKET_FOLDER = \"./dataset/\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import math\n",
+    "import datetime\n",
+    "import time\n",
+    "import glob\n",
+    "\n",
+    "import cudf\n",
+    "import cupy\n",
+    "from numba import cuda\n",
+    "import rmm\n",
+    "\n",
+    "import nvtabular as nvt\n",
+    "from nvtabular.ops import Normalize, FillMissing, Categorify, LogOp, JoinExternal, Dropna, LambdaOp, JoinGroupby, Filter, HashBucket, FillMedian\n",
+    "from nvtabular.column_similarity import ColumnSimilarity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 6 µs, sys: 3 µs, total: 9 µs\n",
+      "Wall time: 14.3 µs\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "from cudf import read_csv\n",
+    "import rmm\n",
+    "rmm.reinitialize(managed_memory=True)  \n",
+    "\n",
+    "documents_meta = read_csv(DATA_BUCKET_FOLDER + 'documents_meta.csv')\n",
+    "train_set = (read_csv(DATA_BUCKET_FOLDER+'clicks_train.csv')\n",
+    "             .merge(read_csv(DATA_BUCKET_FOLDER + 'events.csv'), on=\"display_id\", how=\"left\", suffixes=('', '_event'))\n",
+    "             .merge(read_csv(DATA_BUCKET_FOLDER+'promoted_content.csv'), on=\"ad_id\", how=\"left\", suffixes=('', '_promo'))\n",
+    "             .merge(documents_meta, on=\"document_id\", how=\"left\")\n",
+    "             .merge(documents_meta, left_on=\"document_id_promo\", right_on=\"document_id\", how=\"left\", suffixes=('', \"_promo\"))\n",
+    "            )\n",
+    "train_set.to_parquet(\"train_pdf.parquet\", compression=None)\n",
+    "train_set = None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "documents_categories_cudf = cudf.read_csv(\n",
+    "    DATA_BUCKET_FOLDER + 'documents_categories.csv',\n",
+    "    names=['document_id', 'category_id', 'confidence_level'],\n",
+    "    dtype={'document_id': 'int64','category_id': 'int64','confidence_level': 'float32'},\n",
```

Comment:
Do we need to specify the names? Seems like just using the CSV header would work pretty well (like remove both names and skiprows=1 parameters)

Reply:
Done.